apps/integration/integration/automation/integrate

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+
+# Parse args
+SKIP_BUILD=0
+SKIP_TESTS=0
+SKIP_PUSH=0
+TAG="integration_tests"
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --help)
+      echo "Utility script for building containers, running integration tests, and pushing images"
+      echo "Make sure to run this from the sycamore root directory."
+      echo "-------------------------------------------------------"
+      echo "Arguments:"
+      echo "  --help                Display this message"
+      echo "  --build               Build images"
+      echo "  --tests               Run integration tests"
+      echo "  --push                Push images"
+      echo "  --tag     [TAG]       When building, running, and/or pushing, use this docker tag."
+      echo "                        Default is 'integration_tests'"
+      exit 0
+      ;;
+    --build)
+      SKIP_BUILD=1
+      echo "Will build images"
+      shift
+      ;;
+    --tests)
+      SKIP_TESTS=1
+      echo "Will run integration tests"
+      shift
+      ;;
+    --push)
+      SKIP_PUSH=1
+      echo "Will push images"
+      shift
+      ;;
+    --tag)
+      [[ -z $2 ]] && die "A tag must be speicified when using the --tag arg; e.g. --tag my-tag"
+      [[ $2 == "--*" ]] && die "Detected tag was $2. Tags should not begin with --"
+      TAG="$2"
+      echo "Using tag ${TAG}"
+      shift
+      shift
+      ;;
+  esac
+done
+
+NOW="$(date +"%Y-%m-%d_%H_%M_%S")"
+ARCH="amd64"
+[[ "$(uname -m)" = "arm64" ]] && ARCH="arm64"
+
+RUNDIR="apps/integration/runs/${NOW}"
+GIT_LOGFILE="${RUNDIR}/git.log"
+DOCKER_LOGFILE="${RUNDIR}/docker.log"
+POETRY_LOGFILE="${RUNDIR}/poetry.log"
+PYTEST_LOGFILE="${RUNDIR}/pytest.log"
+QUERY_LOGFILE="${RUNDIR}/test_queries.log"
+
+main() {
+  [[ ! -d ".git" ]] && die "Please run this script from sycamore root!"
+  mkdir -p "${RUNDIR}"
+  echo "Building/testing tag ${TAG}" >&2
+  echo "Get the newest git commits" >&2
+  if checkout_main_if_new; then
+    echo "Changes detected. Running Tests" >&2
+    poetry install --no-root > "${POETRY_LOGFILE}" 2>&1 \
+        && { [[ $SKIP_BUILD ]] || build_images > "${DOCKER_LOGFILE}" 2>&1; } \
+        && { [[ $SKIP_TESTS ]] || runtests > "${PYTEST_LOGFILE}" 2>&1; }
```

Comment:
I'd do:
```
 ...
 && touch "${RUNDIR}/passed_tests"
[[ -f "${RUNDIR}/passed_tests" ]] || touch "${RUNDIR}/failed_tests"
```

If someone put an echo "done with tests" above the local passed_tests=$? line, it would make all the tests appear to pass in the current form.